src/hooks/usePublisher.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useRef, useState } from "react";
+import { Director, Publish } from '@millicast/sdk';
+
+export type PublisherState = "ready" | "connecting" | "streaming";
+
+export interface Publisher {
+    startStreaming: (broadcastOptions: BroadcastOptions) => Promise<void>;
+    stopStreaming: () => void;
+    updateAudioTrack: (mediaStream: MediaStream) => Promise<void>;
+    updateVideoTrack: (mediaStream: MediaStream) => Promise<void>;
```

Comment:
I think it should be
```suggestion
    updateVideoTrack: (track: MediaStreamTrack) => Promise<void>;
```